src/aap_eda/api/serializers/eda_credential.py

```diff
@@ -313,3 +318,28 @@ def get_references(eda_credential: models.EdaCredential) -> list[dict]:
         resources.append(resource)
 
     return resources
+
+
+class EdaCredentialTestSerializer(serializers.ModelSerializer):
+    """Serializer for the Credential Test."""
+
+    metadata = serializers.JSONField(
+        required=False,
+        help_text="Metadata of the credential for testing",
+    )
+
+    def validate(self, attrs):
+        metadata = attrs.get("metadata")
+        inputs = inputs_from_store(self.instance.inputs.get_secret_value())
```

Comment:
Why not call `get_resolved_secrets` function?

Reply:
@hsong-rh When doing a test we can't use resolved secrets, since that is connecting to the external SMS. If the credential is of kind="external" it can be used for tests and it cannot have data coming from the external SMS.